server/src/main/java/io/spine/server/delivery/CatchUpStarter.java

```diff
@@ -0,0 +1,238 @@
+     */
+    public void start(@Nullable Set<I> ids, Timestamp since) throws CatchUpAlreadyStartedException {
+        checkNotActive(ids);
+
+        CatchUp.Request request = buildRequest(ids, since);
+        CatchUpId id = CatchUpId.newBuilder()
+                                .setUuid(Identifier.newUuid())
+                                .setProjectionType(projectionStateType.value())
+                                .vBuild();
+        CatchUpRequested eventMessage = CatchUpRequested
+                .newBuilder()
+                .setId(id)
+                .setRequest(request)
+                .vBuild();
+        EventFactory eventFactory = newEventFactory(projectionStateType, context.isMultitenant());
+        Event event = eventFactory.createEvent(eventMessage, null);
+        context.eventBus()
+               .post(event);
+    }
+
+    private static EventFactory newEventFactory(TypeUrl stateType, boolean multitenant) {
+        String userIdValue = format("`CatchUpStarter` for `%s`", stateType.value());
+        UserId onBehalfOf = UserId.newBuilder()
+                                  .setValue(userIdValue)
+                                  .build();
+        ActorRequestFactory requestFactory = requestFactory(onBehalfOf, multitenant);
+        Any producerId = AnyPacker.pack(onBehalfOf);
+        ActorContext actorContext = requestFactory.newActorContext();
+        return EventFactory.forImport(actorContext, producerId);
+    }
+
+    @SuppressWarnings("MethodWithMultipleLoops")
+    private CatchUp.Request buildRequest(@Nullable Set<I> ids, Timestamp since) {
+        CatchUp.Request.Builder requestBuilder = CatchUp.Request.newBuilder();
+        if (ids != null) {
+            for (I id : ids) {
+                Any packed = Identifier.pack(id);
+                requestBuilder.addTarget(packed);
+            }
+        }
+
+        requestBuilder.setSinceWhen(since);
+        for (EventClass eventClass : eventClasses) {
+            TypeName name = eventClass.typeName();
+            requestBuilder.addEventType(name.value());
+        }
+        return requestBuilder.vBuild();
+    }
+
+    private void checkNotActive(@Nullable Set<I> ids) throws CatchUpAlreadyStartedException {
+        Iterable<CatchUp> ongoing = storage.readByType(projectionStateType);
+        List<CatchUp> active = stream(ongoing)
+                .filter(catchUp -> CatchUpStatus.COMPLETED != catchUp.getStatus())
+                .collect(toList());
+        boolean alreadyCatchingUp = hasIntersections(active, ids);
+        if (alreadyCatchingUp) {
+            throw new CatchUpAlreadyStartedException(projectionStateType, ids);
+        }
+    }
+
+    private static ActorRequestFactory requestFactory(UserId actor, boolean multitenant) {
+        TenantFunction<ActorRequestFactory> function =
+                new TenantFunction<ActorRequestFactory>(multitenant) {
```

Comment:
Please extract into a helper class.
Also, I'm not sure if this is good enough. Can this ever be called from a custom thread (i.e. via an `Executor`)? Maybe, we should pass the tenant ID as a parameter to the builder.